webknossos-datastore/app/com/scalableminds/webknossos/datastore/controllers/ZarrStreamingController.scala

```diff
@@ -265,7 +265,7 @@ class ZarrStreamingController @Inject()(
     for {
       (dataSource, dataLayer) <- dataSourceRepository.getDataSourceAndDataLayer(organizationId,
                                                                                 datasetDirectoryName,
-                                                                                dataLayerName) ~> NOT_FOUND
+                                                                                dataLayerName) ~> BAD_REQUEST
```

Comment:
Hmmm honestly on a conceptual level 404 makes more sense here IMO instead of 400. Is it maybe possible to refill the datasourceRepository upon such an initial request instead? 

Sorry to bump against this fix. Or maybe first add a check whether the datasourceRepository is not yet filled and in case it is still empty return 400?

Moreover, this code runs on the datastore. So it should have all the needed information for the dataset when `organizationId, datasetDirectoryName, dataLayerName` are correct? Thus, I could fill in the needed info on demand, can't it?

Reply:
I agree that 400 is a bit weird. Maybe these requests should return 503 as long as the initialization is not complete? That would cause the zarr libraries to retry.

Reply:
Good Idea, 503 sounds about right. I changed it. I also wrote https://github.com/scalableminds/webknossos/issues/8649 to track avoiding this problem during the normal datastore operation.

Filling in the needed info on demand could also be possible, but ideally not necessary.